src/lib/y2network/interfaces_collection.rb

```diff
@@ -70,6 +71,14 @@ def by_name(name)
       end
     end
 
+    # Returns list of interfaces of given type
+    #
+    # @param type [String] device type name ("eth", "br", ...)
+    # @return [Array<Interfaces>] list of found interfaces
+    def by_type(type)
```

Comment:
What about returning an `InterfacesCollection`? The point is that we could concatenate several calls (`by_type(...).by_whatever(...)`). If you need an array, you could just use a `to_a` method (or implement `first` or whatever you need). It is different from the `by_name` case, because in that case you only expect one interface with a given name.

Reply:
hm, makes sense I'll adapt.